### PR TITLE
Serve the speculative reserve mid-flight once a run passes 30s

### PR DESCRIPTION
**Pull request by holdenk**
The speculative precompute was only ever reachable at the very end of
generate_appeals, after research, make_appeals (up to a 360s budget) and
synthesis had all run. A user whose live generation stalled therefore sat
on a spinner for minutes while a usable appeal was already sitting in the
DB, held back by design.

Add a deadline: SPECULATIVE_FALLBACK_SECONDS (30s), measured from the
start of the flow, after which any run still under ENOUGH_APPEALS flushes
the held-back reserve as it goes. The check runs at the points where the
flow already yields -- after research/enrichment, on each generating-phase
heartbeat, and between streamed drafts -- so a stalled backend costs the
user at most one heartbeat interval past the mark, and a reserve that
lands mid-flight is picked up as soon as it exists.

Rows served this way are promoted to speculative=False exactly as the
end-of-flow reconciliation promotes them. Live drafts are unaffected:
full-context rows still stream as they arrive, so this only ever adds to
what the user gets.

Supporting changes:
- served_texts is now tracked from the top of the flow (existing rows,
  streamed drafts, early flush) instead of being rebuilt from the DB at
  synthesis time, so no path can ship the same text twice -- including a
  draft whose row write failed.
- Early-served reserve rows are excluded from the synthesis inputs by
  text, since promotion would otherwise let the precompute reach the >=2
  synthesis gate that the speculative filter exists to prevent.
- live_new subtracts the early-served count so the zero-appeal
  APPEAL_GEN_DIAG still fires when the reserve rescued a failed run, and
  the done frame reports speculative_appeals.

Co-Authored-By: Claude Opus 5 <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01K39Ke9Z2GNBpLvB3zJSHvJ

## Summary by CodeRabbit

* **New Features**
  * Speculative appeals can now be delivered earlier when generating live appeals takes too long.
  * Streaming results are deduplicated to prevent repeated appeals.
  * Final status information now more accurately reports appeals delivered from the reserve.

* **Bug Fixes**
  * Prevented reserve appeals from being resent during synthesis or final reconciliation.
  * Improved delivery reliability during slow generation and streaming periods.

* **Tests**
  * Added coverage for reserve timing, deduplication, synthesis handling, and reconciliation scenarios.

### CODE REVIEW

File: fighthealthinsurance/common_view_logic.py [hunk 1]
```diff
@@ -3542,6 +3661,11 @@ async def save_appeal(item: GeneratedAppeal) -> dict[str, str]:
                 pass
             passed = time.time() - t
             logger.debug(f"Saved appeal ({len(appeal_text)} chars) in {passed:.1f}s")
+            # Mark it served as soon as it is on its way out, so the early
+            # reserve flush running between streamed drafts can't ship a
+            # speculative row whose text matches one already sent.
+            if appeal_text:
+                served_texts.add(str(appeal_text).strip())
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/common_view_logic.py:3717]: Fixed in 5dd1944. The dedup now happens in the `keep()` filter, before the draft reaches `save_appeal`, so an identical live draft is neither streamed nor counted (nor written as a duplicate row). Tracked as `dupe_count` in the generating-phase diagnostics so an all-duplicate run stays distinguishable from a silent backend, and covered by `test_live_draft_matching_an_early_served_reserve_is_dropped`.

Note the filter runs on the `sync_iterator_to_async` worker thread while the loop may be adding to `served_texts`; a set membership test is a single atomic lookup, and the worst case of racing one is a duplicate slipping through — i.e. today's behavior.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: fighthealthinsurance/common_view_logic.py [hunk 2]
```diff
@@ -3999,10 +4147,15 @@ def keep(item: Optional[GeneratedAppeal]) -> bool:
         models_tried = make_appeals_diag.get("models_tried") or "none"
         # Keyed on live_new (pre-reserve), so a reserve that rescued the user
         # still reports the backend failure. reserve_note says whether the user
-        # was actually left empty-handed or the fallback covered it.
+        # was actually left empty-handed or the fallback covered it -- counting
+        # both routes the reserve can take: the mid-flight deadline flush and
+        # the end-of-flow reconciliation.
+        from_reserve = reconciled + reserve_served
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/common_view_logic.py]: Agreed — fixed in 5dd1944. The reconciliation now tracks `reconciled_from_reserve` alongside `reconciled`, incremented only for rows with speculative provenance (`row.speculative` or `context_level == "speculative"`, sampled before promotion clears the flag, so a reserve row promoted by an earlier run still counts). `served_from_reserve` and the done frame's `speculative_appeals` use that subset; `reconciled` still drives the "late/held-back appeal(s)" line.

`test_late_live_draft_is_not_counted_as_a_speculative_appeal` covers it by landing a full-context row during synthesis — after the flow snapshots what it sent, before the reconciliation reads — so the row is served by the non-mini branch and must report `speculative_appeals == 0`.

---
_Generated by [Claude Code](https://claude.ai/code)_
